### PR TITLE
Upgrade frontend to Vue 2.7

### LINKED ISSUE #1189: Upgrade web app to Vue 2.7
[Vue 2.7 was recently released](https://blog.vuejs.org/posts/vue-2-7-naruto.html) with built-in Composition API, among other things. [The plugin](https://github.com/vuejs/composition-api) we currently have to make use of the Composition API is scheduled to reach EOL by  the end of the year.

The migration path is pretty straightforward and is outlined here https://blog.vuejs.org/posts/vue-2-7-naruto.html, we'll just need to make extra sure that nothing breaks after the upgrade.

### PR DESCRIPTION
**Pull request by mvandenburgh**
Upgrades the frontend to Vue 2.7, which includes the composition API natively. Crucially, this allows us to abandon usage of the [vue composition API plugin](https://github.com/vuejs/composition-api), which will be going EOL at the end of this year. There's also been some major bugs found in that plugin, so I think it makes sense to do this upgrade before diving into some of the console errors and reactivity bugs we've been seeing in the frontend to at least rule it out as the source.

There's other stuff we can and should do at some point, like switching to `<script setup>` syntax and using vite instead of webpack, since presumably that will ease the burden when we upgrade to Vue 3 someday. But for now, this is the bear minimum needed to move to Vue 2.7.

Closes #1189. Requires #1259 to be merged into this branch first.

### CODE REVIEW

**Review by AlmightyYakob — CHANGES_REQUESTED** [review]
Looks good overall, just one main question

File: web/src/components/DandisetsPage.vue
Comment (mvandenburgh) [reply web/src/components/DandisetsPage.vue]: I think you're right, I used the existing `route` variable and it seems to work fine. At one point I was having trouble with reactivity, but it seems likely that something else was causing that.

**Review by AlmightyYakob — APPROVED** [review]
Looks good to me, I think #1259 can be merged into this branch. Any reason this PR is still set as draft?

**Comment by mvandenburgh** [comment]
> Looks good to me, I think #1259 can be merged into this branch. Any reason this PR is still set as draft?

Just as an extra reminder not to merge this before #1259 is merged in. I'll merge #1259 into this branch now, and then merge this branch into `master` once CI is green.

**Pull request merged**
